test/contract_solidity_bytecode_aevm_SUITE.erl

```diff
@@ -0,0 +1,278 @@
+
+
+mock_genesis(Cfg) ->
+    Preset = ?config(preset, Cfg),
+    meck:new(aec_genesis_block_settings, []),
+    meck:expect(aec_genesis_block_settings, preset_accounts, 0, Preset),
+    ok.
+
+unmock_genesis(_Cfg) ->
+    meck:unload(aec_genesis_block_settings),
+    ok.
+
+create_tx(Override, Cfg) ->
+    PubKey = ?config(my_pub_key, Cfg),
+    Code = ?config(code, Cfg),
+    VmVersion = ?config(vm_version, Cfg),
+    Map = #{ owner      => PubKey
+           , nonce      => 1
+           , code       => Code
+           , vm_version => VmVersion
+           , fee        => 10
+           , deposit    => 100
+           , amount     => 0
+           , gas        => 1000000000000000000
+           , gas_price  => 5
+           , call_data  => <<"0">>
+           },
+    Map1 = maps:merge(Map, Override),
+    {ok, Tx} = aect_create_tx:new(Map1),
+    Tx.
+
+
+
+prepare_app_start(App, Config) ->
+    try prepare_app_start_(App, Config)
+    catch
+        error:Reason ->
+            error({Reason, erlang:get_stacktrace()})
+    end.
+
+prepare_app_start_(App, Config) ->
+    application:load(App),
+    TempDir = create_temp_key_dir(),
+    application:set_env(aecore, keys_dir, TempDir),
+    application:set_env(aecore, password, <<"secret">>),
+
+    {ok, Deps0} = application:get_key(App, applications),
+    Deps = maybe_add_mnesia(App, Deps0), % mnesia is started manually in aecore_app
+    AlreadyRunning = [ Name || {Name, _,_} <- proplists:get_value(running_apps, Config) ],
+    [ ok = application:ensure_started(Dep) || Dep <- Deps ],
+    {ok, lists:reverse(Deps -- AlreadyRunning), TempDir}.
+
+app_stop(Apps, TempDir) ->
+    remove_temp_key_dir(TempDir),
+    [ application:stop(App) || App <- Apps ],
+    ok.
+
+maybe_add_mnesia(App, Deps) ->
+    case lists:member(aecore, [App|Deps]) of
+        true  -> Deps ++ [mnesia];
+        false -> Deps
+    end.
+
+remove_temp_key_dir(TmpKeysDir) ->
+    {ok, KeyFiles} = file:list_dir(TmpKeysDir),
+    %% Expect four filenames - private and public keys x2.
+    [_KF1, _KF2, _KF3, _KF4] = KeyFiles,
+    lists:foreach(
+      fun(F) ->
+              AbsF = filename:absname_join(TmpKeysDir, F),
+              {ok, _} = {file:delete(AbsF), {F, AbsF}}
+      end,
+      KeyFiles),
+    ok = file:del_dir(TmpKeysDir).
+
+create_temp_key_dir() ->
+    mktempd(os:type()).
+
+mktempd({unix, _}) ->
+    lib:nonl(?cmd("mktemp -d")).
+
+wait_for_it(Fun, Value) ->
+    wait_for_it(Fun, Value, 0).
+
+wait_for_it(Fun, Value, Sleep) ->
+    case Fun() of
+        Value ->
+            Value;
+        _Other ->
+            %% ?ifDebugFmt("Waiting for ~p got ~p~n",[Value,_Other]),
+            timer:sleep(Sleep),
+            wait_for_it(Fun, Value, Sleep + 10)
+    end.
+
+mock_fast_and_deterministic_cuckoo_pow() ->
+    mock_fast_cuckoo_pow({"mean16s-generic", "", 16}).
+
+mock_fast_cuckoo_pow({_MinerBin, _MinerExtraArgs, _NodeBits} = Cfg) ->
```

Comment:
Could reuse one of aec_test_utils:mock_fast_cuckoo_pow/* ?

Reply:
My view is that the test for one app should not have any dependencies on tests in another app. For one thing I am not sure what "make SUITE=xxx test" will build (why would it build modules in other test dirs?